src/clj/puppetlabs/services/legacy_routes/legacy_routes_core.clj

```diff
@@ -46,21 +85,18 @@
   [handler mount-point api-version request]
   (let [{{environment :environment} :params
          uri :uri
-         query-string               :query-string} request
-        path-info (str "/" (-> (string/split uri #"/" 3) (nth 2)))
-        query-params (if (str/blank? query-string)
-                       {}
-                       (ring-codec/form-decode query-string))]
+         query-string :query-string} request
+        path-info (str "/" (-> (string/split uri #"/" 3) (nth 2)))]
```

Comment:
Now that you pointed that out that smell, I think it stinks. I'll change it real quick.